### PR TITLE
push to pypi

### PR DESCRIPTION
**Pull request by de-code**

### CODE REVIEW

File: project_tests.sh [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #!/bin/bash
```
Comment (de-code) [inline project_tests.sh:1]: Should we switch to hyphen in script names for elife projects as well?

File: Jenkinsfile [hunk 2]
```diff
@@ -1,21 +1,61 @@
...
     }
 
-    stage 'Build image', {
-        sh 'docker build -t elife/sciencebeam-alignment .'
-    }
+    node('containers-jenkins-plugin') {
+        stage 'Build images', {
+            checkout scm
+            dockerComposeBuild(commit)
+            try {
```
Comment (giorgiosironi) [inline Jenkinsfile]: we should extract this as `dockerComposeRun` similar to https://github.com/elifesciences/elife-jenkins-workflow-libs/blob/master/vars/dockerComposeBuild.groovy

File: project_tests.sh (hunk 1 above)
Comment (giorgiosironi) [reply project_tests.sh:1]: Too entrenched to be easily done :stop_sign:

File: Jenkinsfile [hunk 3]
```diff
@@ -1,21 +1,61 @@
...
+            }
+        }
+
+        stage 'Project tests', {
+            try {
+                sh "IMAGE_TAG=${commit} " +
+                    "docker-compose -f docker-compose.yml -f docker-compose.ci.yml " +
+                    "run sciencebeam-alignment ./project_tests.sh"
+            } finally {
+                sh 'docker-compose down -v'
```
Comment (giorgiosironi) [inline Jenkinsfile]: the `docker-compose down` command should still use the same `-f` configuration as the other commands, but we'll solve this with the command extraction

File: project_tests.sh (hunk 1 above)
Comment (de-code) [reply project_tests.sh:1]: Okay, I'll leave it like it is for now.

File: Jenkinsfile [hunk 4]
```diff
@@ -1,21 +1,54 @@
...
     }
 
-    stage 'Build image', {
-        sh 'docker build -t elife/sciencebeam-alignment .'
-    }
+    node('containers-jenkins-plugin') {
+        stage 'Build images', {
+            checkout scm
+            dockerComposeBuild(commit)
+            candidateVersion = dockerComposeRun(
```
Comment (de-code) [inline Jenkinsfile:15]: that seems more readable now with dockerComposeRun

File: Jenkinsfile [hunk 5]
```diff
@@ -1,21 +1,54 @@
...
+                "./project_tests.sh",
+                commit
+            )
+        }
+
+        elifeMainlineOnly {
+            stage 'Push release', {
+                isNew = sh(script: "git tag | grep v${candidateVersion}", returnStatus: true) != 0
+                if (isNew) {
+                    candidateVersion = dockerComposeRun(
```
Comment (giorgiosironi) [inline Jenkinsfile]: does it actually return a new value for `candidateVersion`?

Comment (de-code) [reply Jenkinsfile]: It's shown in Jenkins `Candidate version: v0.0.1`, so seem to work.

Comment (giorgiosironi) [reply Jenkinsfile]: but the `echo` is before this instruction? (which is not executed on the pull request)

Comment (de-code) [reply Jenkinsfile]: Okay, got you. I didn't check where you made this comment. No, that is a copy and paste error. I removed it.

**Review by giorgiosironi — APPROVED** [review]

**Pull request merged**